spec/volcanic/authenticator/v1/warden_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+RSpec.describe Volcanic::Authenticator::V1::Warden, :vcr do
+  before { Configuration.set }
+  let(:mock_request) { double 'request' }
+  let(:allow_always_strategy) { Volcanic::Authenticator::V1::Warden::AllowAlways.new(nil) }
+  let(:validate_token_always_strategy) { Volcanic::Authenticator::V1::Warden::ValidateTokenAlways.new(nil) }
+  let(:validate_token_present_strategy) { Volcanic::Authenticator::V1::Warden::ValidatePresentToken.new(nil) }
+  let(:tokens) { JSON.parse(Configuration.mock_tokens) }
+  let(:mock_token_base64) { tokens['token'] }
+  let(:mock_auth_header) { "Bearer #{mock_token_base64}" }
+  let(:invalid_auth_header) { 'invalid' }
+
+  before do
+    allow(mock_request).to receive(:headers).and_return(mock_auth_header)
+    allow(mock_request).to receive(:has_header?).with('HTTP_AUTHORIZATION').and_return(true)
```

Comment:
if you change this to return `has_header` and add a new `let(:has_header) { true }` then you can simplify some of the below describes, e.g. you could replace the `before` on line 29 with `let(:has_header) { false }`